opm/autodiff/MissingFeatures.cpp

```diff
@@ -17,17 +17,51 @@
   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
 */
 #include <opm/common/OpmLog/OpmLog.hpp>
+
 #include <opm/parser/eclipse/Deck/DeckKeyword.hpp>
 #include <opm/parser/eclipse/Parser/Parser.hpp>
+#include <opm/parser/eclipse/Parser/ParserKeywords/C.hpp>
+#include <opm/parser/eclipse/Parser/ParserKeywords/E.hpp>
+#include <opm/parser/eclipse/Parser/ParserKeywords/P.hpp>
+
 #include <opm/autodiff/MissingFeatures.hpp>
+
 #include <unordered_set>
 #include <string>
-
+#include <map>
+#include <boost/lexical_cast.hpp>
 
 namespace Opm {
 
 namespace MissingFeatures {
 
+
+    template <typename Keyword, typename Item, typename T>
+    void addUnsupported(std::multimap<std::string, PartiallySupported<T> >& map, T itemValue)
+    {
+        std::pair<std::string,PartiallySupported<T> > pair({Keyword::keywordName, PartiallySupported<T>{Item::itemName , itemValue}});
+        map.insert(pair);
+    }
+
+
+    template <typename T>
+    void checkOptions(const DeckKeyword& keyword, std::multimap<std::string , PartiallySupported<T> >& map)
+    {
+        // check for partially supported keywords.
+        typename std::multimap<std::string, PartiallySupported<T> >::iterator it, itlow, itup;
+        itlow = map.lower_bound(keyword.name());
```

Comment:
I'd say that range-for is always better than a manual loop, but (for_each + a lambda) is not always better. In this case, I don't think range-for applies since equal_range unfortunately returns a pair of iterators which is not a valid range (let's hope they fix this for C++20...). So manual iteration between lower_bound and upper_bound is quite ok.